File: leetcode/剑指/剑指.4.二维数组中的查找.go
func findNumberIn2DArray(matrix [][]int, target int) bool {
    if len(matrix) == 0 {
        return false;
    }
    n, m := len(matrix), len(matrix[0]);
    row, col := 0, m - 1;
    for {
        if row >= n || col < 0 {
            break;
        }
        nums := matrix[row][col];
        if nums == target {
            return true;
        } else if nums < target {
            row += 1;
        } else {
            col -= 1;
        }
    }
    return false;
}

File: leetcode/剑指/剑指.4.二维数组中的查找.py
class Solution:
    def findNumberIn2DArray(self, matrix: List[List[int]], target: int) -> bool:
        if(len(matrix) == 0): return False
        n, m = len(matrix), len(matrix[0])
        row, col = 0, m - 1

        while row < n and col >= 0:
            search = matrix[row][col]
            if search == target:
                return True
            elif search < target:
                row += 1
            else:
                col -= 1
        return False